File: DotNetGotchas/CSharp/ValueTypeConstructor/ValueTypeInstantiation/MyType.cs
//MyType.cs
using System;

namespace ValueTypeInstantiation
{
	public struct MyType
	{
		private int val;

		public override string ToString()
		{
			return "The value is " + val;
		}

		//public MyType() {} // Can't be provided

		public MyType(int initialValue)
		{
			val = initialValue;
		}
	}
}

File: DotNetGotchas/VBDN/ValueTypeConstructor/ValueTypeInstantiation/MyType.vb
'MyType.vb

Public Structure MyType
    Private val As Integer

    Public Overrides Function ToString() As String
        Return "The value is " & val
    End Function

    'Public Sub New() ' Can't be provided
    'End Sub

    Public Sub New(ByVal initialValue As Integer)
        val = initialValue
    End Sub
End Structure
